eradiate/scenes/atmosphere/_heterogeneous.py

```diff
@@ -203,19 +190,29 @@ def _sigma_t_fname_validator(self, attribute, value):
     _quantities = {"albedo": "albedo", "sigma_t": "collision_coefficient"}
 
     def __attrs_post_init__(self):
-        # Check for automatic width and height computation
+        # Check attributes consistency
```

Comment:
The only write operations I see are at lines 201-202: if removing them makes sense (which I think is fine, since "auto" is the default value for these fields), then all this code should be transferred to validators.

Reply:
I agree

Reply:
By the way: I am now sure that validators can be shared, so you can attach a single validator to all fields it is supposed to validate (this can include coupled validation). See `ArrayRadProfile` for an example.

Reply:
Thanks for the tip! :)